olive/data/component/dataset.py

```diff
@@ -115,20 +129,32 @@ def __getitem__(self, index):
         if index < 0 or index >= len(self):
             raise IndexError("Index out of range")
 
+        if self.dummy_data[index]:
+            return self.dummy_data[index]
+
         input_types = [resolve_torch_dtype(dtype_str) for dtype_str in self.input_types]
+        generator = torch.Generator().set_state(self.generator_state) if self.generator_state is not None else None
 
-        if not self.input_names:
-            dummy_inputs = []
-            for shape, dtype in zip(self.input_shapes, input_types):
-                dummy_inputs.append(torch.ones(shape, dtype=dtype))
-            dummy_inputs = tuple(dummy_inputs) if len(dummy_inputs) > 1 else dummy_inputs[0]
-        else:
+        # NOTE: torch.rand doesn't support all types, especially not int64 and bool
+        # So, always generate the data with dtype=float32 and convert it to expected input_type.
+        if self.input_names:
             dummy_inputs = {}
             for input_name, input_shape, input_type in zip(self.input_names, self.input_shapes, input_types):
-                dummy_inputs.update({input_name: torch.ones(input_shape, dtype=input_type)})
+                data = torch.rand(input_shape, generator=generator, dtype=torch.float32)
```

Comment:
I think we still need an option or dataset that gives us constant ones for data. The dummy data for pytorch model also uses the dummy data container https://github.com/microsoft/Olive/blob/722749ecaba2117f36474b65b6efb38813ac6b3b/olive/model/handler/mixin/dummy_inputs.py#L28. This might cause issues with the `input_ids` and `attention_masks` being invalid or lead to unexpected behavior.

Reply:
`torch.rand` and then cast to `int` leads to all zeros since the sampling is in `[0,1)`.

Reply:
Also, dummy inputs may be used for model latency evaluation. Should we keep the inputs be consistent among model evaluations like torch, onnx, dlc and etc.?

Reply:
Reverted to use `torch.ones` instead of `torch.rand`. A truly random dataset needs discussion and design. Fallback to what worked in the past for now.